### ISSUE TITLE
[bitnami/kafka] cannot connect to headless svc with port 19092

### ISSUE THREAD

**Issue by NickYadance** [bitnami/charts#18200]
### Name and Version

bitnami/kafka 

### What architecture are you using?

arm64

### What steps will reproduce the bug?
```console
$ helm install kafka oci://registry-1.docker.io/bitnamicharts/kafka

The CLIENT listener for Kafka client connections from within your cluster have been configured with the following security settings:
    - SASL authentication

To connect a client to your Kafka, you need to create the 'client.properties' configuration files with the content below:

security.protocol=SASL_PLAINTEXT
sasl.mechanism=SCRAM-SHA-256
sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required \
    username="user1" \
    password="$(kubectl get secret kafka-user-passwords --namespace default -o jsonpath='{.data.client-passwords}' | base64 -d | cut -d , -f 1)";

To create a pod that you can use as a Kafka client run the following commands:

    kubectl run kafka-client --restart='Never' --image docker.io/bitnami/kafka:3.5.1-debian-11-r7 --namespace default --command -- sleep infinity
    kubectl cp --namespace default /path/to/client.properties kafka-client:/tmp/client.properties
    kubectl exec --tty -i kafka-client --namespace default -- bash

    PRODUCER:
        kafka-console-producer.sh \
            --producer.config /tmp/client.properties \
            --broker-list kafka-controller-0.kafka-controller-headless.default.svc.cluster.local:19092,kafka-controller-1.kafka-controller-headless.default.svc.cluster.local:19092,kafka-controller-2.kafka-controller-headless.default.svc.cluster.local:19092 \
            --topic test

    CONSUMER:
        kafka-console-consumer.sh \
            --consumer.config /tmp/client.properties \
            --bootstrap-server kafka.default.svc.cluster.local:19092 \
            --topic test \
            --from-beginning

$ k get po 
kafka-client         1/1     Running            0                  84m
kafka-controller-0   1/1     Running            0                  6m8s
kafka-controller-1   1/1     Running            0                  6m8s
kafka-controller-2   1/1     Running            0                  6m8s

$ k get svc
NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                         AGE
kafka                       ClusterIP   10.109.128.57    <none>        19092/TCP                       6m22s
kafka-controller-headless   ClusterIP   None             <none>        19094/TCP,19092/TCP,19093/TCP   6m22s
```
failed connecting to svc:19092 in kafka-client
```
I have no name!@kafka-client:/$ kafka-console-producer.sh \
            --producer.config /tmp/client.properties \
            --broker-list kafka-controller-0.kafka-controller-headless.default.svc.cluster.local:19092,kafka-controller-1.kafka-controller-headless.default.svc.cluster.local:19092,kafka-controller-2.kafka-controller-headless.default.svc.cluster.local:19092 \
            --topic test
>[2023-08-04 08:56:35,971] WARN [Producer clientId=console-producer] Connection to node -1 (kafka-controller-0.kafka-controller-headless.default.svc.cluster.local/10.244.12.237:19092) could not be established. Broker may not be available. (org.apache.kafka.clients.NetworkClient)
[2023-08-04 08:56:35,971] WARN [Producer clientId=console-producer] Bootstrap broker kafka-controller-0.kafka-controller-headless.default.svc.cluster.local:19092 (id: -1 rack: null) disconnected (org.apache.kafka.clients.NetworkClient)
[2023-08-04 08:56:35,976] WARN [Producer clientId=console-producer] Connection to node -3 (kafka-controller-2.kafka-controller-headless.default.svc.cluster.local/10.244.3.127:19092) could not be established. Broker may not be available. (org.apache.kafka.clients.NetworkClient)
[2023-08-04 08:56:35,976] WARN [Producer clientId=console-producer] Bootstrap broker kafka-controller-2.kafka-controller-headless.default.svc.cluster.local:19092 (id: -3 rack: null) disconnected (org.apache.kafka.clients.NetworkClient)
```

succeed connecting to 9092 in kafka-client
```
I have no name!@kafka-client:/$ kafka-console-producer.sh             --producer.config /tmp/client.properties             --broker-list kafka-controller-0.kafka-controller-headless.default.svc.cluster.local:9092             --topic test
>
[2023-08-04 08:57:23,457] WARN [Producer clientId=console-producer] Error while fetching metadata with correlation id 4 : {test=UNKNOWN_TOPIC_OR_PARTITION} (org.apache.kafka.clients.NetworkClient)
[2023-08-04 08:57:23,558] WARN [Producer clientId=console-producer] Error while fetching metadata with correlation id 5 : {test=UNKNOWN_TOPIC_OR_PARTITION} (org.apache.kafka.clients.NetworkClient)
[2023-08-04 08:57:23,661] WARN [Producer clientId=console-producer] Error while fetching metadata with correlation id 6 : {test=UNKNOWN_TOPIC_OR_PARTITION} (org.apache.kafka.clients.NetworkClient)
[2023-08-04 08:57:23,763] WARN [Producer clientId=console-producer] Error while fetching metadata with correlation id 7 : {test=UNKNOWN_TOPIC_OR_PARTITION} (org.apache.kafka.clients.NetworkClient)
>12
>13
>14
```

run consumer
```
>I have no name!@kafka-clien:/$ kafka-console-consumer.sh \\
            --consumer.config /tmp/client.properties \
            --bootstrap-server kafka.default.svc.cluster.local:19092 \
            --topic test \
            --from-beginning
12
13
14
```

### Are you using any custom parameters or values?

no

### What do you see instead?

connection failed

**Comment by SergkeiM** [comment]
Hi @NickYadance have the same issue, do you know what is root cause? maybe can submit a PR

**Comment by SergkeiM** [comment]
@joancafom I think because of this v24 is not usable as ‘kafka-controller-0.kafka-controller-headless.default.svc.cluster.local’ doesn’t exists.

**Comment by migruiz4** [comment]
Hi there!

I have submitted this PR fixing this issue: #18274.

On one hand, I noticed I accidentally changed the default values for `service.ports` during one of the tests.
On the other hand, I have modified the value in the notes from `service.ports.client` to `listeners.client.containerPort`.

I'm a bit confused about this one, as I think I misunderstood ports on headless services.

Apparently, when using headless services either in their FQDN format or only service name, the service `port` is not forwarding to the `targetPort` provided, but the containerPort specified in the pod. 

I'm not sure if it is supposed to work like that or if I misconfigured something in the headless service.
At the moment, I will solve this by using the containerPort, which I tested that works, instead of the service port although they are now matching.

In previous versions of the chart, since both 'service.port' and 'containerPort' matched this issue may have been unnoticed.
